fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java

```diff
@@ -0,0 +1,173 @@
+package gov.cms.ab2d.fhir;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.context.FhirVersionEnum;
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.api.EncodingEnum;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.instance.model.api.IBaseBundle;
+import org.hl7.fhir.instance.model.api.IBaseConformance;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+
+@Slf4j
+public enum FhirVersion {
+    STU3("org.hl7.fhir.dstu3.model", FhirContext.forDstu3(), FhirVersionEnum.DSTU3, "/v1/",
+            org.hl7.fhir.dstu3.model.ResourceType.Patient),
+    R4("org.hl7.fhir.r4.model", FhirContext.forR4(), FhirVersionEnum.R4, "/v2/",
+            org.hl7.fhir.r4.model.ResourceType.Patient);
+
+    private String classLocation;
+    private FhirContext context;
+    private IParser jsonParser;
+    private FhirVersionEnum fhirVersionEnum;
+    private String versionString;
+    private Object patientEnum;
+
+    FhirVersion(String classLocation, FhirContext context, FhirVersionEnum fhirVersionEnum, String versionString,
+                Object patientEnum) {
+        this.classLocation = classLocation;
+        this.context = context;
+        this.fhirVersionEnum = fhirVersionEnum;
+        this.versionString = versionString;
+        this.patientEnum = patientEnum;
+    }
+
+    public IParser getJsonParser() {
+        if (this.jsonParser == null) {
+            EncodingEnum respType = EncodingEnum.forContentType(EncodingEnum.JSON_PLAIN_STRING);
+            this.jsonParser = respType.newParser(this.context);
+        }
+        return jsonParser;
+    }
+
+    public static FhirVersion from(FhirVersionEnum fhirVersionEnum) {
+        FhirVersion[] versions = FhirVersion.values();
+        for (FhirVersion v : versions) {
+            if (v.fhirVersionEnum == fhirVersionEnum) {
+                return v;
+            }
+        }
+        return null;
+    }
+
+    public FhirContext getContext() {
+        return this.context;
+    }
+
+    public static FhirVersion fromUrl(String url) {
+        FhirVersion[] versions = FhirVersion.values();
+        for (FhirVersion v : versions) {
```

Comment:
nit: please no 1 letter variable names.

Alternative implementation (still with temporary variable):
return Stream.of(FhirVersion.values().anyMatch(fhir => url.contains(v.versionString));

Reply:
done